azkaban-common/src/main/java/azkaban/executor/container/watch/FlowStatusManagerListener.java

```diff
@@ -239,6 +239,9 @@ private void updatePodStatus(AzPodStatusMetadata event) {
           "Flow execution-id %d for pod %s does not have a final status in database and will be "
               + "finalized.", executionId, event.getPodName()));
       final String reason = "Flow Pod execution was completed.";
+      if (!validateTransition(event)) {
+        executableFlow.setStatus(Status.POD_FAILED);
```

Comment:
Setting the flow status directly will bypass a number of additional tasks performed by `ExecutionControllerUtils.finalizeFlow`, including alerting users about flow failures. 
As mentioned in the previous comment, please see if the `finalizeFlow` can be updated to accept and additional param for `finalStatus`. We can discuss if that seem problematic for any reason.